Error when performing , `yarn start` : 
```
2022-09-28 14:49:14.635 [element-js] ERROR in ../matrix-react-sdk/src/components/views/directory/NetworkDropdown.tsx
2022-09-28 14:49:14.635 [element-js] Module not found: Error: Can't resolve '../../../../../src/util/TchapUtils' in '/Users/mca/Documents/work/projets/betagouv/repo/tchap-web-v4/matrix-react-sdk/src/components/views/directory'
2022-09-28 14:49:14.636 [element-js] ℹ ｢wdm｣: Failed to compile.
2022-09-28 14:49:14.911 [element-js] ℹ ｢wdm｣: Compiling...
2022-09-28 14:49:15.538 [element-js] ✖ ｢wdm｣:    2421 modules
```

--- a/patches/public-room-server-list/matrix-react-sdk+3.54.0.patch
+++ b/patches/public-room-server-list/matrix-react-sdk+3.54.0.patch
@@ -34,7 +34,7 @@ index 701e5e3..e4d8082 100644
  import AccessibleButton from "../elements/AccessibleButton";
  import withValidation from "../elements/Validation";
 -
-+import TchapUtils from "../../../../../src/util/TchapUtils"
++import TchapUtils from "../../../../../../src/util/TchapUtils";
  const SETTING_NAME = "room_directory_servers";
  
  export interface IPublicRoomDirectoryConfig {

--- a/src/util/TchapUtils.ts
+++ b/src/util/TchapUtils.ts
@@ -48,13 +48,14 @@ export default class TchapUtils {
         return s.charAt(0).toUpperCase() + s.slice(1);
     }
 
+    static findHomeServerNameFromUrl = (url: string): string => {
+        const homeServerList = SdkConfig.get()['homeserver_list'];
+        const homeserver = homeServerList.find(homeServer => homeServer.base_url === url);
+        return homeserver.server_name;
+    };
+
     static fetchHomeserverForEmail = async (email: string): Promise<void | {base_url: string, server_name: string}> => {
         const homeServerList = SdkConfig.get()['homeserver_list'];
-
-        const findHomeServerNameFromUrl = (url: string): string => {
-            const homeserver = homeServerList.find(homeServer => homeServer.base_url === url);
-            return homeserver.server_name;
-        };
 
         const randomHomeServer = homeServerList[Math.floor(Math.random() * homeServerList.length)];
         const infoUrl = "/_matrix/identity/api/v1/info?medium=email&address=";
@@ -70,7 +71,7 @@ export default class TchapUtils {
                 const serverUrl = "https://matrix." + response.hs;
                 return {
                     base_url: serverUrl,
-                    server_name: findHomeServerNameFromUrl(serverUrl),
+                    server_name: this.findHomeServerNameFromUrl(serverUrl),
                 };
             })
             .catch((error) => {
@@ -105,8 +106,8 @@ export default class TchapUtils {
      * @returns to "Dinum"
      */
     static toFriendlyServerName = (serverName: string): string => {
-        const friendlyServerName = serverName === "agent.tchap.gouv.fr" ?
-            "Agents" : serverName.replace("agent.", "").replace(".tchap.gouv.fr", "");
+        const serverUrl = "https://matrix." + serverName;
+        const friendlyServerName = this.findHomeServerNameFromUrl(serverUrl);
         return this.capitalize(friendlyServerName);
     };
 }

--- a/src/util/TchapUtils.ts
+++ b/src/util/TchapUtils.ts
@@ -51,7 +51,8 @@ export default class TchapUtils {
      * @returns to "Dinum"
      */
     static toFriendlyServerName = (serverName: string): string => {
-        const friendlyServerName = serverName.replace("agent.", "").replace(".tchap.gouv.fr", "");
+        const friendlyServerName = serverName === "agent.tchap.gouv.fr" ?
+            "Agents" : serverName.replace("agent.", "").replace(".tchap.gouv.fr", "");
         return this.capitalize(friendlyServerName);
     }
 }

